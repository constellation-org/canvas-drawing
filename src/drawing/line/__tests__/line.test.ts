import StraightLine from '../straight'

describe('[Line]', () => {
  let canvas: HTMLCanvasElement
  let context: CanvasRenderingContext2D
  beforeEach(function () {
    canvas = document.createElement('canvas')
    canvas.setAttribute('width', '100')
    canvas.setAttribute('height', '100')
    context = canvas.getContext('2d') as CanvasRenderingContext2D
  })
  it('should draw a solid, dashed and dotted line', () => {
    const solid = new StraightLine({
      style: 'solid',
      from: {
        x: 10,
        y: 20,
      },
      to: {
        x: 90,
        y: 20,
      }
    })
    const dashed = new StraightLine({
      style: 'dashed',
      from: {
        x: 10,
        y: 50,
      },
      to: {
        x: 90,
        y: 50,
      }
    })
    const dotted = new StraightLine({
      style: 'dotted',
      from: {
        x: 10,
        y: 80,
      },
      to: {
        x: 90,
        y: 80,
      }
    })
    solid.draw(context)
    dashed.draw(context)
    dotted.draw(context)
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
  it('should draw a 10 pts width line', () => {
    const line = new StraightLine({
      width: 10,
      from: {
        x: 10,
        y: 20,
      },
      to: {
        x: 90,
        y: 20,
      }
    })
    line.draw(context)
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
  it('should draw a red line', () => {
    const line = new StraightLine({
      color: 'red',
      from: {
        x: 10,
        y: 20,
      },
      to: {
        x: 90,
        y: 20,
      }
    })
    line.draw(context)
    expect(canvas.toDataURL('image/png')).toMatchSnapshot()
  })
})
