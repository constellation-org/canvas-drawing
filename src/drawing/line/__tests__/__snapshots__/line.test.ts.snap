// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Line] should draw a 10 pts width line 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAhBJREFUeF7tkzFuI0EQxKT/P9pOFBtNaEGUAV7MU7XJnferf1MG3lPXdMyrIGMfQUEKMmZg7JxeSEHGDIyd0wspyJiBsXN6IQUZMzB2Ti+kIGMGxs7phRRkzMDYOeSF/Izd/t/OObk+QZ+/vCDffQIn1yeoIN+V+Pzvk+sTVJCCPGJg7EdOH/8J6oU8kvbk+gQVpCCPGBj7kdPHf4J6IY+kPbk+QQXZDPLIVf3I3wbIC8mlYKAggmQyURBiS2ALIkgmEwUhtgS2IIJkMlEQYktgCyJIJhMFIbYEtiCCZDJREGJLYAsiSCYTBSG2BLYggmQyURBiS2ALIkgmEwUhtgS2IIJkMlEQYktgCyJIJhMFIbYEtiCCZDJREGJLYAsiSCYTBSG2BLYggmQyURBiS2ALIkgmEwUhtgS2IIJkMlEQYktgCyJIJhMFIbYEtiCCZDJREGJLYAsiSCYTBSG2BLYggmQyURBiS2ALIkgmEwUhtgS2IIJkMlEQYktgCyJIJhMFIbYEtiCCZDJREGJLYAsiSCYTBSG2BLYggmQyURBiS2ALIkgmEwUhtgS2IIJkMlEQYktgCyJIJhMFIbYEtiCCZDJREGJLYAsiSCYTBSG2BLYggmQyURBiS2ALIkgmEwUhtgS2IIJkMlEQYktgCyJIJhMFIbYEtiCCZDJREGJLYAsiSCYTBSG2BLYggmQy8Qv/FAploEOhUgAAAABJRU5ErkJggg=="`;

exports[`[Line] should draw a red line 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAfxJREFUeF7tmTGKxEAQxMb/f7T9hQEpKIM23hKNtMMF95w+UwaeqWs65hRk7EdQkIKMGRg7pxdSkDEDY+f0QgoyZmDsnF5IQcYMjJ3TCynImIGxc3ohBRkzMHZOL6QgYwbGzumF/DXIe847dvuvznnO3f+erl9IQVj/gjB/+rogulIG1IOwc1rfGrj+G3IL7HvMQEGYP31dEF0pAxaE+dPXBdGVMmBBmD99XRBdKQMWhPnT1wXRlTJgQZg/fV0QXSkDFoT509cF0ZUyYEGYP31dEF0pAxaE+dPXBdGVMmBBmD99XRBdKQMWhPnT1wXRlTJgQZg/fV0QXSkDFoT509cF0ZUyYEGYP31dEF0pAxaE+dPXBdGVMmBBmD99XRBdKQMWhPnT1wXRlTJgQZg/fV0QXSkDFoT509cF0ZUyYEGYP31dEF0pAxaE+dPXBdGVMmBBmD99XRBdKQMWhPnT1wXRlTJgQZg/fV0QXSkDFoT509cF0ZUyYEGYP31dEF0pAxaE+dPXBdGVMmBBmD99XRBdKQMWhPnT1wXRlTJgQZg/fV0QXSkDFoT509cF0ZUyYEGYP31dEF0pAxaE+dPXBdGVMmBBmD99XRBdKQMWhPnT1wXRlTJgQZg/fV0QXSkDFoT509cF0ZUyYEGYP31dEF0pAxaE+dPXBdGVMuAHJd8IZeaFZegAAAAASUVORK5CYII="`;

exports[`[Line] should draw a solid, dashed and dotted line 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAAAXNSR0IArs4c6QAAAmZJREFUeF7tmmFKw2AQBb9eSbySeCXxSuKZlGCrpYJv19Rhi5Nfgb7kbWey+ZXD8hhF4DBqGodZChn2EChEIcMIDBvHDVHIMALDxnFDFDKMwLBx3BCFDCMwbBw3RCHDCAwbxw1RyDACw8ZxQxQyjMCwcdyQGxbyNmz2Wxun9PCXQsd/rpB9j0CJdSmkkH0mjleXWJdCCpkp5CpTeZOfCXQ2RJYAAYUAkDsVCunQArIKASB3KhTSoQVkFQJA7lQopEMLyCoEgNypUEiHFpBVCAC5U6GQDi0gqxAAcqdCIR1aQFYhAOROhUI6tICsQgDInQqFdGgBWYUAkDsVCunQArIKASB3KhTSoQVkFQJA7lQopEMLyCoEgNypUEiHFpDtCKl8bP281no8m3s7fyr8j/u11utZbju/C9dtme2607GdvxS6tpm2OU/Hdv5QuO6S1W94xBqFfMhRyMWj4oYcgbghN7wh8f1nYD+Bzobsb/MOkYBCIiI2oBCWd2xTSETEBhTC8o5tComI2IBCWN6xTSERERtQCMs7tikkImIDCmF5xzaFRERsQCEs79imkIiIDSiE5R3bFBIRsQGFsLxjm0IiIjagEJZ3bFNIRMQGFMLyjm0KiYjYgEJY3rFNIRERG1AIyzu2dYRcfu2dvgb394j/e0AhX0z++oEr6VGIQj4J/LdX2tU3pHRDQ/sIdF5Z+5q8ukRAISVMXEghHOtSk0JKmLiQQjjWpSaFlDBxIYVwrEtNCilh4kIK4ViXmhRSwsSFFMKxLjUppISJCymEY11qUkgJExdSCMe61KSQEiYupBCOdanpHVi3WGXzftsmAAAAAElFTkSuQmCC"`;
